backend-shared/util/log.go

```diff
@@ -1,9 +1,36 @@
 package util
 
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-logr/logr"
+)
+
 const (
 	LogLevel_Debug int = 1
 
 	// LogLevel_Warn is used for unexpected conditions that occur, which might
 	// be an actual error, or might just be the normal functioning of the program.
 	LogLevel_Warn int = -1
 )
+
+type ResourceChangeType string
+
+const (
+	ResourceCreated  ResourceChangeType = "Created"
+	ResourceModified ResourceChangeType = "Modified"
+	ResourceDeleted  ResourceChangeType = "Deleted"
+)
+
+func LogAPIResourceChangeEvent(resourceNamespace string, resourceName string, resource interface{}, resourceChangeType ResourceChangeType, log logr.Logger) {
+
```

Comment:
Hey @jgwest, correct me if I am wrong but  this does not seem to be working here : 
`if log == (logr.Log)(nil)` or  `if log == nil` or `if log == (logr.Logger)(nil)`

error : _invalid operation: cannot compare log == nil (mismatched types logr.Logger and untyped nil)_

Is there anything that needs to be added here or another way to solve this issue?

Reply:
Is this util class specific to App Studio? 
If so, it would help me later if this logging method would add "appstudio-audit":"true" as an additional key-value pair in the JSON struct.

Reply:
@raghavi101 Feel free to ignore my comment on checking that `log == nil`: I thought `log` was an interface (which is a pointer behind the scenes), but just a struct object, so no need to check for nil.

As for @amfred's comment, @raghavi101 can you add add a `log = log.WithValues("audit", "true")` to the `LogAPIResourceChangeEvent` function? That should handle the requirement.